src/QMCWaveFunctions/Jastrow/BsplineFunctor.h

```diff
@@ -140,6 +140,97 @@ struct BsplineFunctor : public OptimizableFunctorBase
                    T* restrict distArrayCompressed,
                    int* restrict distIndices) const;
 
+  static void mw_evaluateVGL(const int iat,
```

Comment:
All the pointers are not required to be dual space but if they are dual space, certain behaviors are expected. Added explanation in the documentation.

Reply:
But don't we have types aliases that are dual space if accelerated device code is compiled?  i.e. Offload...Vector etc.  Then there is actual type safety and the function signature itself will express this.

Reply:
I'm not getting your question.

Reply:
```
//Only OpenMP friendly where there is a hidden link between device and host, all are host pointers
// some are actually dual but necessary to read docs only open mp can still recover device ptr with this API.
function(Pointer* thing
               Pointer* thing2
               int* index
               Pointer* result);

// more informative and easier to understand data dependence provide documentation
// Not workable for omptarget (I think) because the hostptr is used for deviceptr. 
function(Pointer* dev_thing
               Pointer* dev_thing2
               int* index
               Pointer* result);

//Type safe almost leaves function agnostic wrt device implementation.
// Still missing synchronization/resource context argument, there is one if its openmp target
// because its a compiler extension. 
function(DualCont& thing,
               DualCont& thing2,
               int* index,
               Cont& result);
```

Reply:
The class is only meant to OpeMP target offload or host if offload is off. So it is a host function with OpenMP enhancement.
There is no guarantee the OpenMP logic works for any other programming models. So it s not generic dual space but only OpenMP dual space.

I think "dev_thing2" or `DualCont` can lead to miss interpretation.

> // Not workable for omptarget (I think) because the hostptr is used for deviceptr.

it is used as a device_ptr only inside the offload region not at the entry of the function

> // Still missing synchronization/resource context argument, there is one if its openmp target

I think even CUDA doesn't express this and need documentation to help.